src/index.js

```diff
@@ -157,6 +157,10 @@ class UglifyJsPlugin {
       uglifyOptions.output = uglifyOptions.output || {};
     }
 
+    if (typeof this.options.sourceMap === 'undefined' && ['sourcemap', 'source-map', 'hidden-source-map'].indexOf(compiler.options.devtool) > -1) {
```

Comment:
Yes, if we're talking current usage. Because it can be user defined as true or false. This basically only auto-sets the sourceMap option value if u arent manually overriding it